docs/rfcs/0007-synchronized-releases.md

```diff
@@ -0,0 +1,105 @@
+
+## Explanation
+
+### One GeckoView component
+
+Instead of having three components using GeckoView (`browser-engine-gecko-nightly`, `browser-engine-gecko-beta`, `browser-engine-gecko`), we will have just one (`browser-engine-gecko`) and on `master` it will track the latest GeckoView Nightly from `mozilla-central`.
+
+### Releases
+
+Nightly releases of Android Components will come with the latest Nightly version of GeckoView.
+
+For every major version from the Mozilla repositories, we will have a major version release of Android Components. Those versions and releases will be synchronized across our whole stack: Fenix 89 will use Android Components 89, which will use GeckoView 89.
+
+### Branches
+
+On the Android Components side we will continue to have a `master` branch, which will track GeckoView Nightly and from which we will ship Android Components Nightly versions.
+
+For every GeckoView version we will have a matching Android Components release branch (e.g. `releases/89.0`), that we cut from master the day the GeckoView Nightly version becomes Beta. This branch will continue to track the matching GeckoView version from beta builds to release builds.
+
+### Merge day
+
+On merge day the current Nightly version of GeckoView will become the new Beta version. At this point we will cut a matching Android Components release branch, which will continue to track the GeckoView version, while `master` will move on to track the next Nightly version.
+
+Moving code between components will no longer be needed and the merge day procedure will be reduced to branching and versioning, which will be easier to automate.
+
+Example: On the day GeckoView 89 Nightly becomes GeckoView Beta 89, we will cut an Android Components 89 release branch which will track GeckoView Beta 89 (which will eventually become the release version of GeckoView 89). The `master` branch of Android Components will continue to track GeckoView Nightly 90.
+
+### Automation & Bots
+
+We can continue to bump GeckoView and Android Component versions with our bots. The aligned version numbers simplify the process and even allow us to automate releasing and branching too.
+
+* On `master` a bot will continue to update to the latest GeckoView Nightly version. With this change no other version bumps on `master` are needed.
+* On a release branch (e.g. `releases/89.0`) a bot can update to the latest matching GeckoView version. Over time the branch will be bumped from Beta (e.g. 89.0 Beta) to Release versions of GeckoView (e.g. 89.0 release).
+
+Additionally we could automate:
+
+* Whenever the GeckoView Nightly major version changes on merge day and a new GeckoView Beta is available, a bot could cut a matching Android Components release branch and switch to the GeckoView Beta version (e.g. when GeckoView Nightly 89 switches to Nightly 90 a bot could cut a `releases/89.0` branch, bump it to GeckoView Beta 89 and cut a release).
+
```

Comment:
@pocmo Does this make sense?

### Localization

The localization workflow will be very similar to the process in Fenix. We would do the following:

* Frequently import strings to `master` through the `mozilla-l10n-bot`.
* Sync strings from `master` to `releases_v89.0` while `89` is in _Beta_. We would do this up to the merge day.

For the import to `master` no changes are needed. For the string sync between master and the _Beta branch_ we should be able to use the same code as we use for Fenix. (As of this writing, that code is in progress)